documentation/src/main/asciidoc/reference/search-dsl-highlighting.asciidoc

```diff
@@ -0,0 +1,416 @@
+----
+<1> Passing a styled tags schema that will be used to highlight the text.
+====
+
+Using a styled tags schema is just a shortcut to defining the tags as:
+
+.Setting tags as if the styled tags schema is used
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java[tags=fvh-schema-alternative]
+----
+<1> Passing the same collection of tags used when a styled schema is applied.
+====
+
+[NOTE]
+====
+If a styled tags schema is applied to a highlighter, then any other tag configurations are ignored, and the schema tags are used.
+====
+
+[[search-dsl-highlighting-highlighter-options-encoder]]
+=== Encoder
+Encoding can be applied to the highlighted snippets when highlighting the fields that store HTML. Applying an
+HTML encoder to a highlighter will encode the text but keep the highlighting tags as is. By default, a `HighlighterEncoder.DEFAULT`
+encoder is used, which keeps the text as is.
+
+.Setting the HTML encoder
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java[tags=basic-encoder]
+----
+<1> Configuring the HTML encoder.
+====
+
+[NOTE]
+====
+This setting is supported by all highlighter types.
+====
+
+[[search-dsl-highlighting-highlighter-options-no-match-size]]
+=== No match size
+In case of more complex queries or when highlighting is performed for multiple fields, it might lead to a situation
+where the query matched a document, but a particular highlighted field did not contribute to that match.
+This will lead to an empty list of highlights for that particular document and that field.
+No match size configuration allows you to still get some text returned even if the field didn't contribute to the document match,
+and there's nothing to be highlighted in it.
+The number set by this property defines the number of characters to be included starting at the beginning of a field.
+Depending on the highlighter type, the amount of text returned might not precisely match the configured setting since
+highlighters usually try not to break the text in the middle of a word/sentence, depending on their configuration.
+By default, this setting is set to `0` and text will only be returned if there's something to highlight.
+
+.Setting the no match size
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java[tags=no-match-size]
+----
+<1> We are looking for matches in the title.
+<2> In case a word can also be found in the description - we'd want it to be highlighted.
+<3> Setting no match size to `100` to still get at least `100` first characters of a description even if the match for
+a word we are searching for is not found.
+====
+
+[NOTE]
+====
+This setting is supported by all highlighter types.
```

Comment:
Well... it is supported; it's just that the support is limited 😃. I'll replace this note, since we are dropping those anyway, with the one explaining the limitations of this option with Lucene's unified highlighter.